services/transformer/features_impl.go

```diff
@@ -0,0 +1,117 @@
+package transformer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+type featuresService struct {
+	logger   logger.Logger
+	waitChan chan struct{}
+	config   FeaturesServiceConfig
+	features json.RawMessage
+}
+
+func (t *featuresService) SourceTransformerVersion() string {
+	features := t.features
+	if t.features == nil {
+		features = json.RawMessage(defaultTransformerFeatures)
+	}
+
```

Comment:
This is no longer needed since we are initializing the field with the default value in `NewFeaturesService`
```suggestion
```